tests/with_client_cert.yaml

```
@@ -0,0 +1,209 @@
+            -outform PEM
+            -days 3650
+            -nodes
+
+        - name: Generate server certificate
+          block:
+            - name: Set vars for server certificate
+              set_fact:
+                _csr_file: "{{ ara_api_frontend_ssl_cert | regex_replace('.pem$', '.csr') }}"
+
+            - name: Template the openssl configuration file
+              copy:
+                dest: "{{ _selfsigned_conf }}"
+                content: |
+                  [ req ]
+                  default_bits       = 2048
+                  default_keyfile    = {{ ara_api_frontend_ssl_key }}
+                  default_md         = sha256
+                  prompt             = no
+                  distinguished_name = distinguished_name
+                  x509_extensions    = v3_ca
+
+                  [ v3_ca ]
+                  basicConstraints       = critical,CA:FALSE
+                  extendedKeyUsage       = critical,serverAuth
+                  subjectKeyIdentifier   = hash
+                  authorityKeyIdentifier = keyid:always,issuer:always
+
+                  [ distinguished_name ]
+                  commonName             = {{ ara_api_fqdn }}
+                  countryName            = CA
+                  stateOrProvinceName    = QC
+                  localityName           = Montreal
+                  organizationName       = ara
+                  organizationalUnitName = integration-tests
+
+            - name: Generate a CSR
+              command: >-
+                openssl req
+                -config {{ _selfsigned_conf | quote }}
+                -newkey rsa:2048
+                -keyform PEM
+                -out {{ _csr_file | quote }}
+                -outform PEM
+                -nodes
+
+            - name: Template ext file
+              copy:
+                dest: "{{ _selfsigned_ext_conf }}"
+                content: |
+                  subjectAltName=DNS:{{ ara_api_fqdn }}
+
+            - name: Sign the CSR with our CA
+              command: >-
+                openssl x509
+                -req
+                -in {{ _csr_file | quote }}
+                -CA {{ ara_api_frontend_ssl_ca_cert | quote }}
+                -CAkey {{ ara_api_frontend_ssl_ca_key }}
+                -CAcreateserial
+                -out {{ ara_api_frontend_ssl_cert | quote }}
+                -extfile {{ _selfsigned_ext_conf | quote }}
+
+        - name: Generate client certificate
+          block:
+            - name: Set vars for client certificate
+              set_fact:
+                _csr_file: "{{ ara_api_frontend_ssl_cert | regex_replace('.pem$', '.csr') }}"
+
+            - name: Template the openssl configuration file
+              copy:
+                dest: "{{ _selfsigned_conf }}"
+                content: |
+                  [ req ]
+                  default_bits       = 2048
+                  default_keyfile    = {{ ara_api_frontend_ssl_key }}
```

```suggestion
                  default_keyfile    = {{ ara_api_key }}
```

I think this is overwriting NGINX's key with the client's key causing the mismatch.

---

Let's try it, probably my mistake.